### PR TITLE
test: fix ambiguous reference to _1

### PR DESCRIPTION
**Pull request by wjwithagen**
FreeBSD/Clang/lib complains:
/home/jenkins/workspace/ceph-master/src/test/librados_test_stub/TestRadosClient.cc:295:55: error: reference to '_1' is ambiguous
    &TestRadosClient::finish_aio_completion, this, c, _1));
                                                      ^
/usr/local/include/boost/bind/placeholders.hpp:46:38: note: candidate found by name lookup is 'boost::placeholders::_1'
BOOST_STATIC_CONSTEXPR boost::arg<1> _1;
                                     ^
/usr/include/c++/v1/functional:2448:46: note: candidate found by name lookup is 'std::__1::placeholders::_1'
/* _LIBCPP_INLINE_VAR */ constexpr __ph<1>   _1{};

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@wjwithagen could you s/ambigues/ambiguous/ in your commit message?

File: src/test/librados_test_stub/TestRadosClient.cc [hunk 1]
```diff
@@ -292,7 +292,7 @@ void TestRadosClient::flush_aio_operations(AioCompletionImpl *c) {
 int TestRadosClient::aio_watch_flush(AioCompletionImpl *c) {
   c->get();
   Context *ctx = new LambdaContext(std::bind(
-    &TestRadosClient::finish_aio_completion, this, c, _1));
+    &TestRadosClient::finish_aio_completion, this, c, boost::placeholder::_1));
```
Comment (tchaikov) [inline src/test/librados_test_stub/TestRadosClient.cc:295]: this change is wrong, and not necessary.

i added
```c++
using namespace std::placeholders;
```
in 5c1a5bcf18b79e9deb70143f40ae01bbad74444b

**Comment by wjwithagen** [comment]
Right, but it still ends up ambiguous...
So I'll prefix with `std::`??
`std::__1::placeholders::_1`, sort of looks real ugly.

**Comment by tchaikov** [comment]
> Right, but it still ends up ambiguous...
> So I'll prefix with `std::`??
> `std::__1::placeholders::_1`, sort of looks real ugly.

i will try to migrate all `boost::bind` to `std::bind` in a follow-up PR.

File: src/test/librados_test_stub/TestRadosClient.cc [hunk 2]
```diff
@@ -292,7 +292,7 @@ void TestRadosClient::flush_aio_operations(AioCompletionImpl *c) {
 int TestRadosClient::aio_watch_flush(AioCompletionImpl *c) {
   c->get();
   Context *ctx = new LambdaContext(std::bind(
-    &TestRadosClient::finish_aio_completion, this, c, _1));
+    &TestRadosClient::finish_aio_completion, this, c, std::placeholder::_1));
```
Comment (tchaikov) [inline src/test/librados_test_stub/TestRadosClient.cc:295]:
```suggestion
    &TestRadosClient::finish_aio_completion, this, c, std::placeholders::_1));
```

**Comment by tchaikov** [comment]
https://pulpito.ceph.com/kchai-2020-07-30_15:58:21-rados-wip-kefu-testing-2020-07-30-2107-distro-basic-smithi/

**Pull request merged**